dbt/parser/schemas.py

```diff
@@ -61,186 +62,386 @@ def as_kwarg(key, value):
     return "{key}={value}".format(key=key, value=formatted_value)
 
 
-class SchemaParser(BaseParser):
+def build_test_raw_sql(test_namespace, model_name, test_type, test_args):
+    """Build the raw SQL from a test definition.
+
+    :param test_namespace: The test's namespace, if one exists
+    :param model_name: The model name under test
+    :param test_type: The type of the test (unique_id, etc)
+    :param test_args: The arguments passed to the test as a list of `key=value`
+        strings
+    :return: A string of raw sql for the test node.
+    """
+    # sort the dict so the keys are rendered deterministically (for tests)
+    kwargs = [as_kwarg(key, test_args[key]) for key in sorted(test_args)]
+
+    if test_namespace is None:
+        macro_name = "test_{}".format(test_type)
+    else:
+        macro_name = "{}.test_{}".format(test_namespace, test_type)
+
+    raw_sql = "{{{{ {macro}(model=ref('{model}'), {kwargs}) }}}}".format(
+        **{
+            'model': model_name,
+            'macro': macro_name,
+            'kwargs': ", ".join(kwargs)
+        }
+    )
+    return raw_sql
+
 
+class SchemaParser(BaseParser):
+    """This is the original schema parser but with everything in one huge CF of
+    a method so I can refactor it more nicely.
+    """
     @classmethod
-    def parse_schema_test(cls, test_base, model_name, test_config,
-                          test_namespace, test_type, root_project_config,
-                          package_project_config, all_projects, macros=None):
+    def _build_v1_test_args(cls, config):
+        if isinstance(config, (basestring, int, float, bool)):
+            return {'arg': config}
+        else:
+            return config
 
-        if isinstance(test_config, (basestring, int, float, bool)):
-            test_args = {'arg': test_config}
+    # TODO: real exceptions
```

Comment:
👀

Reply:
pretty typical: I long ago resolved this todo and left my comment in. I'll remove the comment.